subsys/bluetooth/shell/bap_broadcast_assistant.c

```diff
@@ -269,10 +270,13 @@ static int cmd_bap_broadcast_assistant_scan_start(const struct shell *sh,
 	int start_scan = false;
 
 	if (argc > 1) {
-		start_scan = strtol(argv[1], NULL, 0);
+		result = 0;
+
+		start_scan = shell_strtobool(argv[1], 0, &result);
+		if (result != 0) {
+			shell_error(sh, "Could not parse start_scan: %d",
+				    result);
 
-		if (start_scan != 0 && start_scan != 1) {
```

Comment:
`shell_strtobool` has different logic, at the end it converts `unsigned long` to `bool`, so any number > 0 is true, right?

Reply:
Right. I was unsure about whether I should just use `shell_strtobool` directly, or actually check if the value was > 1, but decided on the former, as just using `shell_strtobool` would be more consistent with other shell commands that use that.

Reply:
OK, looks reasonable.